kafka-connect-s3/src/main/java/io/confluent/connect/s3/storage/S3OutputStream.java

```diff
@@ -108,9 +111,17 @@ private void uploadPart(int size) throws IOException {
       log.debug("New multi-part upload for bucket '{}' key '{}'", bucket, key);
       multiPartUpload = newMultipartUpload();
     }
-
+    final int partSize = size;
     try {
-      multiPartUpload.uploadPart(new ByteArrayInputStream(buffer.array()), size);
+      if(retry) {
+        RetryWithExponentialBackoff.blocking(new Runnable() {
+          public void run() {
+            multiPartUpload.uploadPart(new ByteArrayInputStream(buffer.array()), partSize);
+          }
+        }, 5, "Part upload failed");
+      } else {
+        multiPartUpload.uploadPart(new ByteArrayInputStream(buffer.array()), partSize);
```

Comment:
I think you can structure this such that it always uses the `RetryWithExponentialBackoff`, no retries would just be the same as `count=1` with the current code.